### ISSUE TITLE
Consider moving training related parameters to `fit()`

### ISSUE THREAD

**Issue by kevinykuo** [DAI-Lab/CTGAN#5]
Things like weight decay, batch size, and epochs may better belong in the `fit` method.
https://github.com/DAI-Lab/CTGAN/blob/94ac76b3ff4dae827eb626abe65923150633e359/ctgan/ctgan_model.py#L426-L432

**Comment by Baukebrenninkmeijer** [comment]
@kevinykuo I second this. No reason for training parameters to be in a different section that the model initialization with said parameters.

**Comment by csala** [comment]
Thanks for the suggestion, @kevinykuo , but I'm afraid I disagree with this.

The rationale behind the parameter organization is:
* `__init__` arguments specify "how" this instance is going to behave, i.e. "how" this model will learn.
* `fit` arguments specify "on which data".

The only argument that could be moved to the `fit` method is `epochs`, and only the code is changed so the creation of the internal instances is not done every time the `fit` method is executed, which would then allow resuming a previous fitting process (see #4 )

All the other arguments are fine where they are and should be kept there.

**Comment by kevinykuo** [comment]
Thanks @csala! I think your comments make sense. A couple of use cases (neither of which hits me currently) that may benefit from moving `batch_size` are 1) one may want to continue training a model on a different machine that doesn't enough GPU memory to support the batch size specified and 2) adaptive batch sizes. This is pretty minor though, so no hearts be broken by a wontfix :)
